### PR TITLE
Consolidate fs usage in utils with fs-extra

### PR DESCRIPTION
**Pull request by dapplion**
Consolidate all `fs` interactions to use `utils/fsUtils`. This uses in turn `fs-extra` which maximizes compatility across versions. The util also adds the feature to path and array of paths and path.join automatically. This helps de-duplicate the common pattern below.

```ts
// Common pattern
import { readFile } from '~/src/utils/fsUtils'
import path from "path"
readFile(path.join("dir", "file.ext"))


// With this PR
import { readFile } from '~/src/utils/fsUtils'
readFile(["dir", "file.ext"])
```

### CODE REVIEW

File: src/tasks/start-task.ts [hunk 1]
```diff
@@ -66,8 +65,15 @@ ${accountsStr}`)
...
     const config: AragonConfig = bre.config.aragon as AragonConfig
-    const appExists = await _checkApp(config.appSrcPath as string)
-    if (appExists) {
+    if (!config.appSrcPath) {
+      logMain(
+        'Warning: appSrcPath is not defined, will continue development without building any front end.'
+      )
+    } else if (!pathExists(config.appSrcPath)) {
+      logMain(
+        `Warning: No front end found ${config.appSrcPath}, will continue development without building any front end.`
```
Comment (sohkai) [inline src/tasks/start-task.ts]:
```suggestion
        `Warning: No front end found at ${config.appSrcPath}, will continue development without building any front end.`
```

File: src/utils/fsUtils.ts [hunk 2]
```diff
@@ -1,82 +1,90 @@
...
  * ```js
- * readFileIfExists("dir", "file.ext")
+ * readFileIfExists(["dir", "file.ext"])
  * ```
  */
-export const readFileIfExists = (...filepaths: string[]): string | undefined =>
-  pathExists(...filepaths) ? readFile(...filepaths) : undefined
+export const readFileIfExists = (
+  filepath: string | string[]
+): string | undefined => (pathExists(filepath) ? readFile(filepath) : undefined)
```
Comment (sohkai) [inline src/utils/fsUtils.ts]: This for me gets a bit hard to read, with all the typings.

Maybe I'm just not used to it, but what do you think about splitting it up so it has a brace and `return` statement?

**Comment by dapplion** [comment]
> LGTM; I just wonder if there are other places where this pattern for `fs` paths is used.
> 
> I do think removing the `path.join()` is elegant, but it may also be unintuitive or confusing for those used to the normal `fs` module.

I also agree with you, while this reduces duplication it may look foreign to other which is an important property. I would skip this PR for now since there are other more pressing issues to solve now

File: src/tasks/start/frontend/aragon-client.ts [hunk 3]
```diff
@@ -54,13 +53,13 @@ export async function installAragonClientIfNeeded(
...
-  if (!fs.existsSync(path.resolve(clientPath))) {
+  if (!pathExists(clientPath)) {
     return true
   }
 
-  if (!fs.existsSync(path.resolve(clientPath, 'build/index.html'))) {
+  if (!pathExists(path.resolve(clientPath, 'build/index.html'))) {
     logFront('Malformed client detected, removing it for re-installation.')
-    await fsExtra.remove(clientPath)
+    remove(clientPath)
```
Comment (sohkai) [inline src/tasks/start/frontend/aragon-client.ts:62]: Sanity checking, but should we do an `await` on the `remove()`? What if this fails—do we handle the promise ever rejecting?

**Pull request merged**